python/tvm/relay/frontend/common.py

```diff
@@ -938,9 +939,14 @@ def ensure_scalar_shape(x):
         return x
 
     num_elem = np.prod(x_shape)
-    assert num_elem == 1, "Cannot squeeze tensor shape {} to scalar form.".format(x_shape)
-
-    return _op.squeeze(x)
+    if num_elem == 1:
+        return _op.squeeze(x)
+    else:
+        if force_assert:
+            assert num_elem == 1, "Cannot squeeze tensor shape {} to scalar form.".format(x_shape)
+        else:
+            return x
```

Comment:
This changes the behavior of `ensure_scalar_shape` -- now we can return non-scalar elements, I would instead in `QLinearMatMul's` `try_resolve_to_const` simply, wrap `ensure_scalar_shape` calls with a check if the number of elements is 1.